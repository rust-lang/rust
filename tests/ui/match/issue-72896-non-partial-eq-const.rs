trait EnumSetType {
    type Repr;
}

enum Enum8 { }
impl EnumSetType for Enum8 {
    type Repr = u8;
}

#[derive(PartialEq, Eq)]
struct EnumSet<T: EnumSetType> {
    __enumset_underlying: T::Repr,
}

const CONST_SET: EnumSet<Enum8> = EnumSet { __enumset_underlying: 3 };

fn main() {
    match CONST_SET {
        CONST_SET => { /* ok */ } //~ ERROR constant of non-structural type `EnumSet<Enum8>` in a pattern
        _ => panic!("match fell through?"),
    }
}
