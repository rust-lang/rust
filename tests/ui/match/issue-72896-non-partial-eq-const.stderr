error: constant of non-structural type `EnumSet<Enum8>` in a pattern
  --> $DIR/issue-72896-non-partial-eq-const.rs:19:9
   |
LL | enum Enum8 { }
   | ---------- must be annotated with `#[derive(PartialEq)]` to be usable in patterns
...
LL | const CONST_SET: EnumSet<Enum8> = EnumSet { __enumset_underlying: 3 };
   | ------------------------------- constant defined here
...
LL |         CONST_SET => { /* ok */ }
   |         ^^^^^^^^^ constant of non-structural type
   |
   = note: see https://doc.rust-lang.org/stable/std/marker/trait.StructuralPartialEq.html for details

error: aborting due to 1 previous error

